articles/create-a-simple-rest-api-application-using-typescript-and-nodejs/index.md

```diff
@@ -0,0 +1,370 @@
+
+![The most beloved language](/engineering-education/create-a-simple-rest-api-application-using-typescript-and-nodejs/the-most-beloved-language.png)
+
+[Image source](https://insights.stackoverflow.com/survey/2020#technology-most-loved-dreaded-and-wanted-languages-loved)
+
+Just like JavaScript, it supports numerous frameworks such as Node.js.
+
+### Goal
+
+The main advantage of using both Typescript and Node.js together is to take the full benefits that this two have. Typescript is well known for supporting Static Typing to detect code errors during typing. It helps you catch errors at an early stage of development, creating a faster application development pipeline, unlike Node.js (JavaScript), where you only detect errors during the application runtime. Typescript is a good choice when building extensive applications.
+
+On the side, Node.js is well known for its diverse open-source libraries, multi-threading ability, Asynchronous, and Event-Driven code execution, being fast and light, etc.
+
+Check out these two guides and learn more about the popularity two.
+
+- [Why is Node.js wildly popular among developers?](/engineering-education/typescript-static-typing/)
+- [Why Static Typing & Why is Typescript so popular?](/engineering-education/why-node-js-is-popular/)
+
+This article will explain how to use Typescript with Node.js Framework. We'll create a simple REST API using Typescript and  Node.js libraries to demonstrate this. The API implements some common web-based API methods such as GET, POST, DELETE, PUT, AND PATCH. We will use a post-application. The application will consume a [Free mock-up API](https://jsonplaceholder.typicode.com/posts) hosted in this [JSON placeholder server](https://jsonplaceholder.typicode.com/).
+
+### Prerequisites
+
+- Ensure that you have [Node.js](https://nodejs.org/en/) and [Postman](https://www.postman.com/) installed on your computer.
+- Essential experience of how to write and execute [Typescript](/engineering-education/a-friendly-beginner-guide-to-typescript/).
+- Be familiar with [Node.js](/engineering-education/history-of-nodejs/) and how to use libraries such as [Express.js](/engineering-education/express/).
+- Be familiar with how to [test APIs using Postman](https://www.youtube.com/watch?v=KFuaybrXCdw), (a tool for interacting with web-based APIs).
+
+### Installing Typescript
+
+Unlike JavaScript, Typescript doesn't run directly on the browser. To execute any Typescript written code, you need a Typescript compiler. This will compile Typescript into JavaScript. This way, it will be easier to executed and run the Typescripts on a browser. Since we are using Node.js, we will install Typescript from NPM using the command below.
+
+```bash
+npm install -g Typescript
+```
+
+Adding the `–g` flag to install the packages globally. That way, Typescript is available to any Node.js project.
+
+### A step by step guide
+
```

Comment:
Remove this heading.  Jump right into the steps using `###` .  

With this heading gone, instead of using `**bold**` to create small headings, use `####`. 

For example, where you have `1. **Fetch all posts**`, you can now have `#### 1. Fetch all posts`